// @guildedjs/api/dist/generated/router/services/ChannelsService.js
"use strict";
/* istanbul ignore file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelsService = void 0;
class ChannelsService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Create a channel
     * Only server channels are supported at this time (coming soon™: DM Channels!)
     * @returns any Success
     * @throws ApiError
     */
    channelCreate({ requestBody, }) {
        return this.httpRequest.request({
            method: "POST",
            url: "/channels",
            body: requestBody,
            mediaType: "application/json",
        });
    }
    /**
     * Get a channel
     * Only server channels are supported at this time (coming soon™: DM Channels!)
     * @returns any Success
     * @throws ApiError
     */
    channelRead({ channelId }) {
        return this.httpRequest.request({
            method: "GET",
            url: "/channels/{channelId}",
            path: {
                channelId: channelId,
            },
        });
    }
    /**
     * Update a channel
     * Only server channels are supported at this time (coming soon™: DM Channels!)
     * @returns any Success
     * @throws ApiError
     */
    channelUpdate({ channelId, requestBody, }) {
        return this.httpRequest.request({
            method: "PATCH",
            url: "/channels/{channelId}",
            path: {
                channelId: channelId,
            },
            body: requestBody,
            mediaType: "application/json",
        });
    }
    /**
     * Delete a channel
     * Only server channels are supported at this time (coming soon™: DM Channels!)
     * @returns void
     * @throws ApiError
     */
    channelDelete({ channelId }) {
        return this.httpRequest.request({
            method: "DELETE",
            url: "/channels/{channelId}",
            path: {
                channelId: channelId,
            },
        });
    }
}
exports.ChannelsService = ChannelsService;
