// @guildedjs/api/dist/generated/router/services/UserStatusService.js
"use strict";
/* istanbul ignore file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserStatusService = void 0;
class UserStatusService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Update your status
     * **Note** - at this time, you can only set a status on your own user
     * @returns void
     * @throws ApiError
     */
    userStatusCreate({ userId, requestBody, }) {
        return this.httpRequest.request({
            method: "PUT",
            url: "/users/{userId}/status",
            path: {
                userId: userId,
            },
            body: requestBody,
            mediaType: "application/json",
        });
    }
    /**
     * Delete your status
     * **Note** - at this time, you can only set a status on your own user
     * @returns void
     * @throws ApiError
     */
    userStatusDelete({ userId, }) {
        return this.httpRequest.request({
            method: "DELETE",
            url: "/users/{userId}/status",
            path: {
                userId: userId,
            },
        });
    }
}
exports.UserStatusService = UserStatusService;
