// @guildedjs/api/dist/generated/router/schemas/$Announcement.js
"use strict";
/* istanbul ignore file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.$Announcement = void 0;
/* eslint-disable */
exports.$Announcement = {
    properties: {
        id: {
            type: "string",
            description: `The ID of the announcement`,
            isRequired: true,
        },
        serverId: {
            type: "string",
            description: `The ID of the server`,
            isRequired: true,
        },
        channelId: {
            type: "string",
            description: `The ID of the channel`,
            isRequired: true,
            format: "uuid",
        },
        createdAt: {
            type: "string",
            description: `The ISO 8601 timestamp that the announcement was created at`,
            isRequired: true,
            format: "date-time",
        },
        createdBy: {
            type: "string",
            description: `The ID of the user who created this announcement`,
            isRequired: true,
        },
        content: {
            type: "string",
            description: `The content of the announcement`,
            isRequired: true,
            format: "markdown",
        },
        mentions: {
            type: "Mentions",
        },
        title: {
            type: "string",
            description: `The title of the announcement`,
            isRequired: true,
            maxLength: 128,
            minLength: 1,
        },
    },
};
