// @guildedjs/api/dist/generated/router/schemas/$CalendarEventCommentReaction.js
"use strict";
/* istanbul ignore file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.$CalendarEventCommentReaction = void 0;
/* eslint-disable */
exports.$CalendarEventCommentReaction = {
    properties: {
        channelId: {
            type: "string",
            description: `The ID of the channel`,
            isRequired: true,
            format: "uuid",
        },
        createdBy: {
            type: "string",
            description: `The ID of the user who added the reaction`,
            isRequired: true,
        },
        emote: {
            type: "Emote",
            isRequired: true,
        },
        calendarEventId: {
            type: "number",
            description: `The ID of the calendar event`,
            isRequired: true,
            minimum: 1,
        },
        calendarEventCommentId: {
            type: "number",
            description: `The ID of the calendar event comment`,
            isRequired: true,
            minimum: 1,
        },
    },
};
